### ISSUE TITLE
TypeError('sequence item 0: expected string, NoneType found',)

### ISSUE THREAD

**Issue by eddified** [harelba/q#126]
Running q against my specific input file (using any simple query) results in this strange error. The error does not seem to be caused by any particular row -- deleting either the first _or_ the last data row (i.e. not including the header), or any row in between, causes the error to no longer occur. The entire file, however, when used as input, will result in this error message. Removing just the header will continue to reproduce the issue.

Tested on Mac El Capitan with unix line endings.

```
$ q -d , -H 'select count(*) from test.csv'
TypeError('sequence item 0: expected string, NoneType found',)
```

Tested against 1.5 _and_ master as of today -- same results.
[test.csv.zip](https://github.com/harelba/q/files/332172/test.csv.zip)

**Comment by harelba** [comment]
@eddified Sorry, missed that issue a while back. Probably not relevant anymore, but just for completeness and for other people. 

The root cause of this is that the last line contains a malformed field content in terms of double-quoting - Seems that the `rm_html_file_name` field contains a partial json, and hence partial escaping of double-quotes. This causes q to misdetect the column types and the number of columns (detecting 86 columns for this row instead of 81). When deleting other lines from the file, q's autodetection changes (since the autodetection is according to the actual data), and this is the reason why the error was gone/changed.

Here's the content of the partial field (row with `1334565323` in it):
```
"4|{""ad_body"": ""Special savings on our online marketing tools & custom services to get you up & running."", ""ad_id"": ""6010875463996"", ""viewtag"": ""https://altfarm.mediaplex.com/ad/ck/14302-167341-17235-399?mpt=[CACHEBUSTER]"">Click Here"""""
```

This problem in turn leads to exposing a bug in q, related to "relaxed mode" string concatenation handling. I've fixed this in the master branch, and pushed it.

**Comment by eddified** [comment]
Thanks!

**Comment by harelba** [comment]
Fixed in v1.6.3.
